packages/arb-checkpointer/checkpointing/indexedCheckpointer.go

```diff
@@ -293,15 +300,56 @@ func deleteCheckpointForKey(bs machine.BlockStore, db machine.CheckpointStorage,
 }
 
 type restoreContextLocked struct {
-	db machine.CheckpointStorage
+	db       machine.CheckpointStorage
+	values   map[common.Hash]value.Value
+	machines map[common.Hash]machine.Machine
 }
 
-func (rcl *restoreContextLocked) GetValue(h common.Hash) value.Value {
-	return rcl.db.GetValue(h)
+func newRestoreContextLocked(db machine.CheckpointStorage, manifest *ckptcontext.CheckpointManifest) (*restoreContextLocked, error) {
+	rcl := restoreContextLocked{db, map[common.Hash]value.Value{}, map[common.Hash]machine.Machine{}}
+
+	valueCache, err := cmachine.NewValueCache()
```

Comment:
I think it would probably be cleaner to just make the value cache a member of restoreContextLocked since the lifetimes seem like a natural fit for each other and then we could remove the `vc` argument from the two getter functions which would let us simplify their callsites a bunch